packages/flutter/test/widgets/binding_live_test.dart

```diff
@@ -0,0 +1,55 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+// This file is for tests for WidgetsBinding that require a `LiveTestWidgetsFlutterBinding`
+void main() {
+  LiveTestWidgetsFlutterBinding();
+  testWidgets('ReportTiming callback records the sendFramesToEngine when it was scheduled', (WidgetTester tester) async {
+    // This test needs LiveTestWidgetsFlutterBinding for multiple reasons.
+    // First, this was the environment that this bug was discovered.
+    // Second, LiveTestWidgetsFlutterBinding doesn't override
+    // scheduleWarmUpFrame, so that the test can start with no frames rendered
+    // allowing deferFirstFrame to work (see below).
```

Comment:
I don't understand this paragraph. In my observation, the problem is that the warm-up frame triggered internally by `testWidgets` before the test body executes is actually send to the engine, but after that `resetFirstFrame` is called, which now puts engine and framework in an inconsistent state: The framework now things no frame has been sent (which means `deferFirstFrame` does have an effect) while the engine things the first frame has been sent.

Reply:
Kind of. The `_runTestBody` has:
```dart
    runApp(Container(key: UniqueKey(), child: _preTestMessage)); // Reset the tree to a known state.
    await pump();
```
so a frame is pumped to the engine anyway, warm-up or not. 

And your understanding on the inconsistency is absolutely correct. Now there's a reportTiming callback pending, which is supposed to only happen after rendered frames, while the framework thinks no frames have been rendered, allowing `deferFirstFrame` to take effect. This combined situation of `LiveTestWidgetsFlutterBinding` and `testWidgets` is probably the only case that can trigger this bug.